protocol/jsm/v2/subscriber.go

```diff
@@ -0,0 +1,38 @@
+/*
+ Copyright 2021 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package jsm
+
+import (
+	"github.com/nats-io/nats.go"
+)
+
+// The Subscriber interface allows us to configure how the subscription is created
+type Subscriber interface {
+	Subscribe(jsm nats.JetStreamContext, subject string, cb nats.MsgHandler) (*nats.Subscription, error)
```

Comment:
in nats.go Subscribe is a variadic that has a few options `nats.SubOpt`, so that would make the consumer become a durable by passing the `nats.Durable("name")` option for example.

https://github.com/nats-io/nats.go/blob/c7fc3c7fdf53ba0fef1a28400253d7f8d2c1a037/js.go#L116

As it is right now the consumers would consumer from JetStream but would be ephemeral so cannot resume from where it was on restart.

Reply:
done @wallyqs